metaflow/plugins/airflow/compute/k8s.py

```diff
@@ -0,0 +1,189 @@
+    KUBERNETES_CONTAINER_IMAGE,
+    BATCH_METADATA_SERVICE_HEADERS,
+    DATASTORE_CARD_S3ROOT,
+)
+
+
+def create_k8s_args(
+    datastore,
+    metadata,
+    environment,
+    flow_name,
+    run_id,
+    step_name,
+    task_id,
+    attempt,
+    code_package_url,
+    code_package_sha,
+    step_cli,
+    docker_image,
+    service_account=None,
+    secrets=None,
+    node_selector=None,
+    namespace=None,
+    cpu=None,
+    gpu=None,
+    disk=None,
+    memory=None,
+    run_time_limit=timedelta(days=5),
+    retries=None,
+    retry_delay=None,
+    env={},
+    user=None,
+):
+
+    k8s = Kubernetes(
+        datastore, metadata, environment, flow_name, run_id, step_name, task_id, attempt
+    )
+    labels = {
+        "app": "metaflow",
+        "metaflow/flow_name": sanitize_label_value(flow_name),
+        "metaflow/step_name": sanitize_label_value(step_name),
+        "app.kubernetes.io/name": "metaflow-task",
+        "app.kubernetes.io/part-of": "metaflow",
+        "app.kubernetes.io/created-by": sanitize_label_value(user),
+    }
+    # Add Metaflow system tags as labels as well!
+    for sys_tag in metadata.sticky_sys_tags:
+        labels["metaflow/%s" % sys_tag[: sys_tag.index(":")]] = sanitize_label_value(
+            sys_tag[sys_tag.index(":") + 1 :]
+        )
+
+    additional_mf_variables = {
+        "METAFLOW_CODE_SHA": code_package_sha,
+        "METAFLOW_CODE_URL": code_package_url,
+        "METAFLOW_CODE_DS": datastore.TYPE,
+        "METAFLOW_USER": user,
+        "METAFLOW_SERVICE_URL": BATCH_METADATA_SERVICE_URL,
+        "METAFLOW_SERVICE_HEADERS": json.dumps(BATCH_METADATA_SERVICE_HEADERS),
+        "METAFLOW_DATASTORE_SYSROOT_S3": DATASTORE_SYSROOT_S3,
+        "METAFLOW_DATATOOLS_S3ROOT": DATATOOLS_S3ROOT,
+        "METAFLOW_DEFAULT_DATASTORE": "s3",
+        "METAFLOW_DEFAULT_METADATA": "service",
+        "METAFLOW_KUBERNETES_WORKLOAD": 1,
+        "METAFLOW_RUNTIME_ENVIRONMENT": "kubernetes",
+        "METAFLOW_CARD_S3ROOT": DATASTORE_CARD_S3ROOT,
+    }
+    env.update(additional_mf_variables)
+    k8s_operator_args = dict(
+        namespace=namespace,
+        service_account=service_account,
+        # todo : pass secrets from metaflow to Kubernetes via airflow
+        secrets=secrets,
+        node_selector=node_selector,
+        cmds=k8s._command(
+            code_package_url=code_package_url,
+            step_cmds=[step_cli],
+        ),
+        in_cluster=True,
+        image=docker_image,
+        cpu=cpu,
+        memory=memory,
+        disk=disk,
+        execution_timeout=dict(seconds=run_time_limit.total_seconds()),
+        retry_delay=dict(seconds=retry_delay.total_seconds()) if retry_delay else None,
+        retries=retries,
+        env_vars=[dict(name=k, value=v) for k, v in env.items()],
+        labels=labels,
+        is_delete_operator_pod=True,
+    )
+    return k8s_operator_args
+
+
+# todo : below code block only for reference. Remove it later.
```

Comment:
The entire code block below this will be removed. It is only here for checking argument types.